model/src/main/java/org/openremote/model/security/KeyStoreService.java

```diff
@@ -0,0 +1,85 @@
+package org.openremote.model.security;
+
+import org.openremote.model.Container;
+import org.openremote.model.ContainerService;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.TrustManagerFactory;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.util.logging.Logger;
+
+public interface KeyStoreService extends ContainerService {
+	String OR_KEYSTORE_PASSWORD = "OR_KEYSTORE_PASSWORD";
+	Logger LOG = Logger.getLogger(KeyStoreService.class.getName());
+
+	KeyStore getKeyStore(String realm, KeyStoreType type);
+	void storeKeyStore(KeyStore keyStore, String realm, KeyStoreType type);
+	char[] getKeyStorePassword();
+
+	KeyManagerFactory getKeyManagerFactory(String realm, String preferredAlias) throws Exception;
```

Comment:
I thought we discussed not exposing the realm to the key factory but instead the key alias should be prefixed with the realm name (i.e. in MQTTProtocol where the client is created).